### PR TITLE
Parse Permit Url

### LINKED ISSUE #526: Parse permit URL
Right now [here](https://github.com/ubiquity/ubiquibot/blob/aeb5956cfdbbb0d05942f2aa3c0eb5e4126eea09/src/handlers/payout/action.ts#L149) we use a plain string comparison to check that the comment is a permit URL. This approach is brittle.

We should parse URL with [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) in order to not rely on query parameters order.

### PR DESCRIPTION
**Pull request by Sadaf-A**
Resolves #526

### CODE REVIEW

File: src/handlers/payout/action.ts [hunk 1]
```diff
@@ -189,7 +189,14 @@ export const handleIssueClosed = async () => {
   logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
   const comment =
     `#### Task Assignee Reward\n### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
-  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot);
+  const permitComments = comments.filter((content) => {
+    if (content.body.includes("https://pay.ubq.fi") && content.user.type == UserType.Bot) {
```
Comment (whilefoo) [inline src/handlers/payout/action.ts:193]: use `permitBaseUrl` from the config and extract this into helper function and use it every where we search for permits

**Review by whilefoo — CHANGES_REQUESTED** [review]
Also post QA

File: src/handlers/payout/action.ts (hunk 1 above)
Comment (Sadaf-A) [reply src/handlers/payout/action.ts]: @whilefoo Hi, please confirm what should be done here

Comment (whilefoo) [reply src/handlers/payout/action.ts]: No you don't need to add `?claim`

File: src/handlers/payout/action.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ import { bountyInfo } from "../wildcard";
 import Decimal from "decimal.js";
 import { GLOBAL_STRINGS } from "../../configs";
 import { isParentIssue } from "../pricing";
+import { DEFAULT_PERMIT_BASE_URL } from "../../configs";
```
Comment (whilefoo) [inline src/handlers/payout/action.ts:22]: not like this, you need to use `getBotConfig`. also post QA

File: src/handlers/payout/action.ts [hunk 3]
```diff
@@ -189,7 +190,14 @@ export const handleIssueClosed = async () => {
   logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
   const comment =
     `#### Task Assignee Reward\n### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
-  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot);
+  const permitComments = comments.filter((content) => {
```
Comment (whilefoo) [inline src/handlers/payout/action.ts:193]: extract this into helper function and apply it everywhere where we search for permit

**Review by whilefoo — CHANGES_REQUESTED** [review]

**Review by 0xcodercrane — APPROVED** [review]

File: src/handlers/payout/action.ts (hunk 3 above)
Comment (0xcodercrane) [reply src/handlers/payout/action.ts:193]: If we make it as a helper fn, that would be cool but not so important for now. To speed up the velocity, mark it as resolved

**Review by 0xcodercrane — APPROVED** [review]

**Comment by whilefoo** [comment]
the parsing should be done with URLSearchParams to avoid dependence on param order like @Sadaf-A did. I don't understand why you changed it to regex

**Comment by Sadaf-A** [comment]
> the parsing should be done with URLSearchParams to avoid dependence on param order like @Sadaf-A did. I don't understand why you changed it to regex

@whilefoo 
@0xcodercrane made those changes so I didn't wanna make any more changes i fixed the errors that arised after it

**Comment by whilefoo** [comment]
> @whilefoo
> @0xcodercrane made those changes so I didn't wanna make any more changes i fixed the errors that arised after it earlier i was also using URLSearchParams as requested by you

yes I know @0xcodercrane made those changes

**Comment by 0xcodercrane** [comment]
We don't create a complex claim URL here, so I guess no need to worry about the param order.

In upper lines, we were also using regex so wanted to make the parse consistent. 

If the order problem happens, lets update it at that time.

**Comment by pavlovcik** [comment]
This isn't the spec

**Comment by Sadaf-A** [comment]
> the parsing should be done with URLSearchParams to avoid dependence on param order like @Sadaf-A did. I don't understand why you changed it to regex

@whilefoo @pavlovcik should i open another PR for this?

**Pull request merged**